### PR TITLE
Bring analyzer fixes to master

### PR DESCRIPTION
**Pull request by TimHess**

### CODE REVIEW

File: src/CircuitBreaker/src/Hystrix.MetricsStreamCore/HystrixApplicationBuilderExtensions.cs [hunk 1]
```diff
@@ -18,7 +18,7 @@ public static IApplicationBuilder UseHystrixMetricsStream(this IApplicationBuild
                 throw new ArgumentNullException(nameof(builder));
             }
 
-            var service = builder.ApplicationServices.GetRequiredService<RabbitMetricsStreamPublisher>();
+            _ = builder.ApplicationServices.GetRequiredService<RabbitMetricsStreamPublisher>();
```
Comment (jkonicki) [inline src/CircuitBreaker/src/Hystrix.MetricsStreamCore/HystrixApplicationBuilderExtensions.cs:21]: Is this correct?

File: src/CircuitBreaker/src/HystrixBase/ExecutionResult.cs [hunk 2]
```diff
@@ -137,14 +137,14 @@ public bool ContainsAnyOf(BitArray other)
...
-                    case HystrixEventType.FALLBACK_EMIT: return _numFallbackEmissions;
-                    case HystrixEventType.EXCEPTION_THROWN: return ContainsAnyOf(EXCEPTION_PRODUCING_EVENTS) ? 1 : 0;
-                    case HystrixEventType.COLLAPSED: return _numCollapsed;
-                    default: return Contains(eventType) ? 1 : 0;
-                }
+                    HystrixEventType.EMIT => _numEmissions,
+                    HystrixEventType.FALLBACK_EMIT => _numFallbackEmissions,
+                    HystrixEventType.EXCEPTION_THROWN => ContainsAnyOf(EXCEPTION_PRODUCING_EVENTS) ? 1 : 0,
+                    HystrixEventType.COLLAPSED => _numCollapsed,
+                    _ => Contains(eventType) ? 1 : 0,
```
Comment (jkonicki) [inline src/CircuitBreaker/src/HystrixBase/ExecutionResult.cs:146]: underscore as variable?

File: src/CircuitBreaker/src/HystrixBase/ExecutionResult.cs [hunk 3]
```diff
@@ -515,15 +515,15 @@ public override string ToString()
...
-                case HystrixEventType.TIMEOUT: return true;
-                case HystrixEventType.CANCELLED: return true;
-                default: return false;
-            }
+                HystrixEventType.SUCCESS => true,
+                HystrixEventType.FAILURE => true,
+                HystrixEventType.BAD_REQUEST => true,
+                HystrixEventType.TIMEOUT => true,
+                HystrixEventType.CANCELLED => true,
+                _ => false,
```
Comment (jkonicki) [inline src/CircuitBreaker/src/HystrixBase/ExecutionResult.cs:525]: underscore as variable?

File: src/CircuitBreaker/src/HystrixBase/HystrixEventTypeHelper.cs [hunk 4]
```diff
@@ -47,86 +47,51 @@ static HystrixEventTypeHelper()
...
+                HystrixEventType.FALLBACK_EMIT => false,
+                HystrixEventType.FALLBACK_SUCCESS => false,
+                HystrixEventType.FALLBACK_FAILURE => false,
+                HystrixEventType.FALLBACK_REJECTION => false,
+                HystrixEventType.FALLBACK_MISSING => false,
+                HystrixEventType.EXCEPTION_THROWN => false,
+                HystrixEventType.RESPONSE_FROM_CACHE => false,
+                HystrixEventType.CANCELLED => false,
+                HystrixEventType.COLLAPSED => false,
+                _ => throw new ArgumentOutOfRangeException(),
```
Comment (jkonicki) [inline src/CircuitBreaker/src/HystrixBase/HystrixEventTypeHelper.cs:69]: underscore as variable name?

File: src/CircuitBreaker/src/HystrixBase/HystrixEventTypeHelper.cs [hunk 5]
```diff
@@ -47,86 +47,51 @@ static HystrixEventTypeHelper()
...
+                HystrixRollingNumberEvent.FALLBACK_EMIT => HystrixEventType.FALLBACK_EMIT,
+                HystrixRollingNumberEvent.FALLBACK_SUCCESS => HystrixEventType.FALLBACK_SUCCESS,
+                HystrixRollingNumberEvent.FALLBACK_FAILURE => HystrixEventType.FALLBACK_FAILURE,
+                HystrixRollingNumberEvent.FALLBACK_REJECTION => HystrixEventType.FALLBACK_REJECTION,
+                HystrixRollingNumberEvent.FALLBACK_MISSING => HystrixEventType.FALLBACK_MISSING,
+                HystrixRollingNumberEvent.EXCEPTION_THROWN => HystrixEventType.EXCEPTION_THROWN,
+                HystrixRollingNumberEvent.RESPONSE_FROM_CACHE => HystrixEventType.RESPONSE_FROM_CACHE,
+                HystrixRollingNumberEvent.COLLAPSED => HystrixEventType.COLLAPSED,
+                HystrixRollingNumberEvent.BAD_REQUEST => HystrixEventType.BAD_REQUEST,
+                _ => throw new ArgumentOutOfRangeException("Not an event that can be converted to HystrixEventType : " + @event),
```
Comment (jkonicki) [inline src/CircuitBreaker/src/HystrixBase/HystrixEventTypeHelper.cs:93]: underscore as variable name?

File: src/CircuitBreaker/src/HystrixBase/Metric/CachedValuesHistogram.cs [hunk 6]
```diff
@@ -103,37 +103,37 @@ public int GetMean()
...
+                8500 => _p85,
+                9000 => _p90,
+                9500 => _p95,
+                9900 => _p99,
+                9950 => _p99_5,
+                9990 => _p99_9,
+                9995 => _p99_95,
+                9999 => _p99_99,
+                10000 => _p100,
+                _ => throw new ArgumentException("Percentile (" + percentile + ") is not currently cached"),
```
Comment (jkonicki) [inline src/CircuitBreaker/src/HystrixBase/Metric/CachedValuesHistogram.cs:135]: underscore as variable name?

File: src/CircuitBreaker/src/HystrixBase/ExecutionResult.cs (hunk 2 above)
Comment (TimHess) [reply src/CircuitBreaker/src/HystrixBase/ExecutionResult.cs:146]: https://docs.microsoft.com/en-us/dotnet/csharp/discards

**Pull request merged**